akka-http-tests/src/test/scala/akka/http/scaladsl/server/IntegrationRoutingSpec.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.http.scaladsl.server
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.{ Http, TestUtils }
+import akka.http.scaladsl.client.RequestBuilding
+import akka.http.scaladsl.model.{ HttpResponse, HttpRequest }
+import akka.stream.ActorMaterializer
+import org.scalatest.concurrent.{ IntegrationPatience, ScalaFutures }
+import org.scalatest.{ BeforeAndAfterAll, Matchers, WordSpecLike }
+import scala.concurrent.duration._
+import scala.concurrent.Await
+
+/** INTERNAL API - not (yet?) ready for public consuption */
+private[akka] trait IntegrationRoutingSpec extends WordSpecLike with Matchers with BeforeAndAfterAll
+  with Directives with RequestBuilding
+  with ScalaFutures with IntegrationPatience {
+
+  implicit val system = ActorSystem(getClass.getSimpleName + "-System")
```

Comment:
is this name correct? we do crazy things in AkkaSpec to get at the class name from a constructor, but perhaps it is different for a `val`?